[BUG] Apply on ovh_cloud_project_kube_nodepool crashes

### Describe the bug
A clear and concise description of what the bug is.

### Terraform Version

```
❯ terraform -v
Terraform v1.6.2
on darwin_amd64
+ provider registry.terraform.io/ovh/ovh v0.34.0
```

### OVH Terraform Provider Version

```
❯ terraform init

Initializing the backend...

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Reusing previous version of ovh/ovh from the dependency lock file
- Using previously-installed ovh/ovh v0.34.0

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

### Affected Resource(s)

Please list the resources as a list, for example:
- ovh_cloud_project_kube_nodepool

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    ovh = {
      source  = "ovh/ovh"
      version = "~> 0.34.0"
    }
  }
  required_version = ">= 1.6.2"
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-7"
  flavor_name    = "b2-7"
  desired_nodes  = var.control_plane_b2_7_node_count
  anti_affinity  = false
  monthly_billed = false
  template {
    spec {
      unschedulable = false
      taints = [
        {
          effect = "PreferNoSchedule"
        }
      ]
    }
    metadata {
      annotations = {}
      finalizers  = ["ovhcloud.com/v1beta1", "ovhcloud.com/v1"]
      labels = {
        "node-restriction.kubernetes.io/flavor-range"   = "b2"
        "node-restriction.kubernetes.io/instance-type"  = "b2-7"
        "node-restriction.kubernetes.io/public-network" = "250M"
        "node-restriction.kubernetes.io/role"           = "controle-plane"
      }
    }
  }
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-30"
  flavor_name    = "b2-30"
  desired_nodes  = var.control_plane_b2_30_node_count
  anti_affinity  = false
  monthly_billed = false
  template {
    spec {
      unschedulable = false
      taints = [
        {
          effect = "PreferNoSchedule"
        }
      ]
    }
    metadata {
      annotations = {}
      finalizers  = ["ovhcloud.com/v1beta1", "ovhcloud.com/v1"]
      labels = {
        "node-restriction.kubernetes.io/flavor-range"   = "b2"
        "node-restriction.kubernetes.io/instance-type"  = "b2-30"
        "node-restriction.kubernetes.io/public-network" = "500M"
        "node-restriction.kubernetes.io/role"           = "controle-plane"
      }
    }
  }
}

```

### Expected Behavior

The resources are created without issues.

### Actual Behavior

The provider crashes

```
Stack trace from the terraform-provider-ovh_v0.34.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 116 [running]:
github.com/ovh/terraform-provider-ovh/ovh.loadNodelPoolTemplateFromResource({0x19d0480, 0xc0008b0440})
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:201 +0x9d4
github.com/ovh/terraform-provider-ovh/ovh.(*CloudProjectKubeNodePoolCreateOpts).FromResource(0xc000882320, 0x22?)
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:87 +0x24e
github.com/ovh/terraform-provider-ovh/ovh.resourceCloudProjectKubeNodePoolCreate(0x0?, {0x191a240?, 0xc0001aa510})
        github.com/ovh/terraform-provider-ovh/ovh/resource_cloud_project_kube_nodepool.go:242 +0x18c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1b4e2c0?, {0x1b4e2c0?, 0xc0006cede0?}, 0xd?, {0x191a240?, 0xc0001aa510?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000472c40, {0x1b4e2c0, 0xc0006cede0}, 0xc00030d450, 0xc000015e80, {0x191a240, 0xc0001aa510})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0003e6e88, {0x1b4e2c0?, 0xc0006cecc0?}, 0xc000796460)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000340000, {0x1b4e2c0?, 0xc0006ce360?}, 0xc0002682a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x19d0320?, 0xc000340000}, {0x1b4e2c0, 0xc0006ce360}, 0xc000268230, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001d61e0, {0x1b51db8, 0xc000007ba0}, 0xc00089a480, 0xc0004aae70, 0x20a08c0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc0001d61e0, {0x1b51db8, 0xc000007ba0}, 0xc00089a480, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-ovh_v0.34.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-11-01T22:16:05.007+0100 [DEBUG] provider: plugin exited
```

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Additional context

Without specifying the template block the plugin does not crashes, example:

```hcl
resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-7"
  flavor_name    = "b2-7"
  desired_nodes  = var.control_plane_b2_7_node_count
  anti_affinity  = false
  monthly_billed = false
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-30"
  flavor_name    = "b2-30"
  desired_nodes  = var.control_plane_b2_30_node_count
  anti_affinity  = false
  monthly_billed = false
}
```

Here is another example of a panic on 0.28.1
```
2023-11-03T17:48:31.325+0100 [INFO]  provider.terraform-provider-ovh_v0.28.1: 2023/11/03 17:48:31 [DEBUG] Logged in on OVH API: timestamp="2023-11-03T17:48:31.325+0100"
ovh_cloud_project_kube_nodepool.control_plane_b2_30_node: Modifying... [id=825ad91c-5df2-47b3-a9a2-7e5ab6f363be]
2023-11-03T17:48:31.340+0100 [INFO]  Starting apply for ovh_cloud_project_kube_nodepool.control_plane_b2_30_node
ovh_cloud_project_kube_nodepool.control_plane_b2_7_node: Modifying... [id=d9eb8208-4a80-4ea5-96dc-d0f9c66dcbc8]
2023-11-03T17:48:31.340+0100 [INFO]  Starting apply for ovh_cloud_project_kube_nodepool.control_plane_b2_7_node
ovh_cloud_project_kube_nodepool.control_plane_b2_15_node: Modifying... [id=f693da73-7997-41ad-9169-6b3c926c084a]
2023-11-03T17:48:31.340+0100 [INFO]  Starting apply for ovh_cloud_project_kube_nodepool.control_plane_b2_15_node
2023-11-03T17:48:31.341+0100 [DEBUG] ovh_cloud_project_kube_nodepool.control_plane_b2_30_node: applying the planned Update change
2023-11-03T17:48:31.341+0100 [DEBUG] ovh_cloud_project_kube_nodepool.control_plane_b2_7_node: applying the planned Update change
2023-11-03T17:48:31.341+0100 [DEBUG] ovh_cloud_project_kube_nodepool.control_plane_b2_15_node: applying the planned Update change
2023-11-03T17:48:31.344+0100 [INFO]  provider.terraform-provider-ovh_v0.28.1: 2023/11/03 17:48:31 [DEBUG] Will update nodepool: ovh.CloudProjectKubeNodePoolUpdateOpts{Autoscale:(*bool)(0xc00089ef58), DesiredNodes:(*int)(0xc00089ef60), MaxNodes:(*int)(0xc00089ef68), MinNodes:(*int)(nil), Template:(*ovh.CloudProjectKubeNodePoolTemplate)(0xc000692190)}: timestamp="2023-11-03T17:48:31.344+0100"
2023-11-03T17:48:31.348+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: panic: runtime error: index out of range [0] with length 0
2023-11-03T17:48:31.348+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1
2023-11-03T17:48:31.348+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: goroutine 99 [running]:
2023-11-03T17:48:31.348+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/ovh/terraform-provider-ovh/ovh.loadNodelPoolTemplateFromResource(0x1cab260, 0xc00071c4c0, 0x8, 0x1cab260, 0xc00071c4c0)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:123 +0x14c5
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/ovh/terraform-provider-ovh/ovh.(*CloudProjectKubeNodePoolUpdateOpts).FromResource(0xc000327aa0, 0xc000a9ed00, 0xc00054d6d0, 0x3, 0x3)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:219 +0x18a
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/ovh/terraform-provider-ovh/ovh.resourceCloudProjectKubeNodePoolUpdate(0xc000a9ed00, 0x1bc7220, 0xc0006e00e0, 0x2, 0xffffffffffffffff)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/ovh/terraform-provider-ovh/ovh/resource_cloud_project_kube_nodepool.go:305 +0x2a5
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000304000, 0x1ea8548, 0xc000924880, 0xc000a9ed00, 0x1bc7220, 0xc0006e00e0, 0x0, 0x0, 0x0)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:363 +0x1ee
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000304000, 0x1ea8548, 0xc000924880, 0xc000460dd0, 0xc000594d00, 0x1bc7220, 0xc0006e00e0, 0x0, 0x0, 0x0, ...)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:475 +0x390
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001aa5e8, 0x1ea8548, 0xc000924880, 0xc0008a9b80, 0xc000924880, 0x1ca44e0, 0xc00093e000)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:977 +0xacf
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0007921a0, 0x1ea85f0, 0xc000924880, 0xc000259110, 0xc0007921a0, 0xc00093e060, 0xc00084bba0)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0xb5
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ca44e0, 0xc0007921a0, 0x1ea85f0, 0xc00093e060, 0xc000904c00, 0x0, 0x1ea85f0, 0xc00093e060, 0xc000940000, 0x933)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100, 0xc00031e3f0, 0x244dbc0, 0x0, 0x0, 0x0)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: google.golang.org/grpc.(*Server).handleStream(0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100, 0x0)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0003905d0, 0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100)
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   google.golang.org/grpc@v1.32.0/server.go:859 +0xab
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-11-03T17:48:31.349+0100 [DEBUG] provider.terraform-provider-ovh_v0.28.1:   google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
2023-11-03T17:48:31.351+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/ovh/ovh/0.28.1/darwin_amd64/terraform-provider-ovh_v0.28.1 pid=35024 error="exit status 2"
2023-11-03T17:48:31.351+0100 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-11-03T17:48:31.351+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-11-03T17:48:31.351+0100 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-11-03T17:48:31.351+0100 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-11-03T17:48:31.376+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-11-03T17:48:31.376+0100 [ERROR] vertex "ovh_cloud_project_kube_nodepool.control_plane_b2_7_node" error: Plugin did not respond
2023-11-03T17:48:31.399+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-11-03T17:48:31.399+0100 [ERROR] vertex "ovh_cloud_project_kube_nodepool.control_plane_b2_30_node" error: Plugin did not respond
2023-11-03T17:48:31.423+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-11-03T17:48:31.423+0100 [ERROR] vertex "ovh_cloud_project_kube_nodepool.control_plane_b2_15_node" error: Plugin did not respond
╷
│ Error: Plugin did not respond
│
│   with ovh_cloud_project_kube_nodepool.control_plane_b2_7_node,
│   on control-plane.tf line 1, in resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node":
│    1: resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with ovh_cloud_project_kube_nodepool.control_plane_b2_15_node,
│   on control-plane.tf line 31, in resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_15_node":
│   31: resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_15_node" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with ovh_cloud_project_kube_nodepool.control_plane_b2_30_node,
│   on control-plane.tf line 61, in resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node":
│   61: resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-ovh_v0.28.1 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 99 [running]:
github.com/ovh/terraform-provider-ovh/ovh.loadNodelPoolTemplateFromResource(0x1cab260, 0xc00071c4c0, 0x8, 0x1cab260, 0xc00071c4c0)
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:123 +0x14c5
github.com/ovh/terraform-provider-ovh/ovh.(*CloudProjectKubeNodePoolUpdateOpts).FromResource(0xc000327aa0, 0xc000a9ed00, 0xc00054d6d0, 0x3, 0x3)
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:219 +0x18a
github.com/ovh/terraform-provider-ovh/ovh.resourceCloudProjectKubeNodePoolUpdate(0xc000a9ed00, 0x1bc7220, 0xc0006e00e0, 0x2, 0xffffffffffffffff)
        github.com/ovh/terraform-provider-ovh/ovh/resource_cloud_project_kube_nodepool.go:305 +0x2a5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000304000, 0x1ea8548, 0xc000924880, 0xc000a9ed00, 0x1bc7220, 0xc0006e00e0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:363 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000304000, 0x1ea8548, 0xc000924880, 0xc000460dd0, 0xc000594d00, 0x1bc7220, 0xc0006e00e0, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:475 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001aa5e8, 0x1ea8548, 0xc000924880, 0xc0008a9b80, 0xc000924880, 0x1ca44e0, 0xc00093e000)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:977 +0xacf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0007921a0, 0x1ea85f0, 0xc000924880, 0xc000259110, 0xc0007921a0, 0xc00093e060, 0xc00084bba0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1ca44e0, 0xc0007921a0, 0x1ea85f0, 0xc00093e060, 0xc000904c00, 0x0, 0x1ea85f0, 0xc00093e060, 0xc000940000, 0x933)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100, 0xc00031e3f0, 0x244dbc0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0003905d0, 0xc0002ad340, 0x1eb25d8, 0xc0001b7380, 0xc00076e100)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ovh_v0.28.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-11-03T17:48:31.451+0100 [DEBUG] provider: plugin exited
```

--- a/ovh/resource_cloud_project_kube_nodepool.go
+++ b/ovh/resource_cloud_project_kube_nodepool.go
@@ -1,6 +1,7 @@
 package ovh
 
 import (
+	"errors"
 	"fmt"
 	"log"
 	"strings"
@@ -203,6 +204,25 @@ func resourceCloudProjectKubeNodePool() *schema.Resource {
 										Elem: &schema.Schema{
 											Type: schema.TypeMap,
 											Set:  schema.HashString,
+											ValidateFunc: func(taintInterface interface{}, path string) (warning []string, errorList []error) {
+												taint := taintInterface.(map[string]interface{})
+
+												if taint["key"] == nil {
+													return nil, []error{errors.New(fmt.Sprintf("key attribute is mandatory for taint: %s", path))}
+												}
+
+												if taint["effect"] == nil {
+													return nil, []error{errors.New(fmt.Sprintf("effect attribute is mandatory for taint: %s", path))}
+												}
+
+												effectString := taint["effect"].(string)
+												effect := TaintEffecTypeToID[effectString]
+												if effect == NotATaint {
+													return nil, []error{fmt.Errorf("effect: %s is not a allowable taint %#v", effectString, TaintEffecTypeToID)}
+												}
+
+												return nil, nil
+											},
 										},
 									},
 								},

--- a/ovh/resource_cloud_project_kube_nodepool_test.go
+++ b/ovh/resource_cloud_project_kube_nodepool_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"regexp"
 	"strings"
 	"testing"
 	"time"
@@ -11,6 +12,12 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+var (
+	effectTaintsErrorRegex       = regexp.MustCompile("(.)*effect attribute is mandatory for taint(.)*")
+	keyTaintsErrorRegex          = regexp.MustCompile("(.)*key attribute is mandatory for taint(.)*")
+	valueNoCrashTaintsErrorRegex = regexp.MustCompile("(.)*This service does not exist(.)*")
 )
 
 func init() {
@@ -72,6 +79,105 @@ func testSweepCloudProjectKubeNodePool(region string) error {
 	}
 	return nil
 }
+
+var testAccCloudProjectKubeNodePoolConfigEffectMissingInTaint = `
+resource "ovh_cloud_project_kube_nodepool" "pool" {
+  service_name  = "xxx"
+  kube_id       = "xxx"
+  name          = "xxx"
+  flavor_name   = "b2-7"
+  desired_nodes = 1
+  min_nodes     = 0
+  max_nodes     = 1
+  template {
+    metadata {
+      annotations = {
+        a1 = "av1"
+      }
+      finalizers = ["finalizer.extensions/v1beta1"]
+      labels = {
+        l1 = "lv1"
+      }
+    }
+    spec {
+      unschedulable = false
+      taints = [
+        {
+          #effect = "PreferNoSchedule"
+          key    = "t1"
+          value  = "tv1"
+        }
+      ]
+    }
+  }
+}
+`
+
+var testAccCloudProjectKubeNodePoolConfigKeyMissingInTaint = `
+resource "ovh_cloud_project_kube_nodepool" "pool" {
+  service_name  = "xxx"
+  kube_id       = "xxx"
+  name          = "xxx"
+  flavor_name   = "b2-7"
+  desired_nodes = 1
+  min_nodes     = 0
+  max_nodes     = 1
+  template {
+    metadata {
+      annotations = {
+        a1 = "av1"
+      }
+      finalizers = ["finalizer.extensions/v1beta1"]
+      labels = {
+        l1 = "lv1"
+      }
+    }
+    spec {
+      unschedulable = false
+      taints = [
+        {
+          effect = "PreferNoSchedule"
+          #key    = "t1"
+          value  = "tv1"
+        }
+      ]
+    }
+  }
+}
+`
+
+var testAccCloudProjectKubeNodePoolConfigValueMissingInTaint = `
+resource "ovh_cloud_project_kube_nodepool" "pool" {
+  service_name  = "xxx"
+  kube_id       = "xxx"
+  name          = "xxx"
+  flavor_name   = "b2-7"
+  desired_nodes = 1
+  min_nodes     = 0
+  max_nodes     = 1
+  template {
+    metadata {
+      annotations = {
+        a1 = "av1"
+      }
+      finalizers = ["finalizer.extensions/v1beta1"]
+      labels = {
+        l1 = "lv1"
+      }
+    }
+    spec {
+      unschedulable = false
+      taints = [
+        {
+          effect = "PreferNoSchedule"
+          key    = "t1"
+          #value  = "tv1"
+        }
+      ]
+    }
+  }
+}
+`
 
 var testAccCloudProjectKubeNodePoolConfig = `
 resource "ovh_cloud_project_kube" "cluster" {
@@ -293,6 +399,53 @@ func TestAccCloudProjectKubeNodePool(t *testing.T) {
 					kubernetesClusterID := state.RootModule().Resources["ovh_cloud_project_kube.cluster"].Primary.ID
 					return fmt.Sprintf("%s/%s/%s", os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST"), kubernetesClusterID, poolId), nil
 				},
+			},
+		},
+	})
+}
+
+func TestAccCloudProjectKubeNodePoolTaints(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckCloud(t)
+			testAccCheckCloudProjectExists(t)
+			testAccPreCheckKubernetes(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCloudProjectKubeNodePoolConfigEffectMissingInTaint,
+				ExpectError: effectTaintsErrorRegex,
+			},
+		},
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckCloud(t)
+			testAccCheckCloudProjectExists(t)
+			testAccPreCheckKubernetes(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCloudProjectKubeNodePoolConfigKeyMissingInTaint,
+				ExpectError: keyTaintsErrorRegex,
+			},
+		},
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckCloud(t)
+			testAccCheckCloudProjectExists(t)
+			testAccPreCheckKubernetes(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCloudProjectKubeNodePoolConfigValueMissingInTaint,
+				ExpectError: valueNoCrashTaintsErrorRegex,
 			},
 		},
 	})

--- a/ovh/types_cloud_project_kube_nodepool.go
+++ b/ovh/types_cloud_project_kube_nodepool.go
@@ -196,11 +196,15 @@ func loadNodelPoolTemplateFromResource(i interface{}) (*CloudProjectKubeNodePool
 					return nil, fmt.Errorf("effect: %s is not a allowable taint %#v", effectString, TaintEffecTypeToID)
 				}
 
-				template.Spec.Taints = append(template.Spec.Taints, Taint{
+				taintObject := Taint{
 					Effect: effect,
 					Key:    taint.(map[string]interface{})["key"].(string),
-					Value:  taint.(map[string]interface{})["value"].(string),
-				})
+				}
+				if taint.(map[string]interface{})["value"] != nil {
+					taintObject.Value = taint.(map[string]interface{})["value"].(string)
+				}
+
+				template.Spec.Taints = append(template.Spec.Taints, taintObject)
 			}
 
 			// spec.unschedulable

--- a/website/docs/r/cloud_project_kube_nodepool.html.markdown
+++ b/website/docs/r/cloud_project_kube_nodepool.html.markdown
@@ -80,7 +80,10 @@ The following arguments are supported:
         * `finalizers` - Finalizers to apply to each node. A finalizer name must be fully qualified, e.g. kubernetes.io/pv-protection , where you prefix it with hostname of your service which is related to the controller responsible for the finalizer.
         * `labels` - Labels to apply to each node
     * `spec` - Spec of each node in the pool
-        * `taints` - Taints to apply to each node
+        * `taints` - Taints to apply to each node [NodeSpec kubernetes documentation](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/node-v1/#NodeSpec)
+          * `effect` - mandatory possible values: NoExecute, NoSchedule, PreferNoSchedule
+          * `key` - mandatory
+          * `value` - (Optional)
         * `unschedulable` - If true, set nodes as un-schedulable
 
 ## Attributes Reference

